test/ng/locationSpec.js

```diff
@@ -2480,6 +2480,10 @@ describe('$location', function() {
       expect(parseLinkAndReturn(locationUrl, 'someIgnoredAbsoluteHref', '#test')).toEqual('http://server/pre/otherPath#test');
     });
 
+    it('should cope with double slashes in the path', function() {
+      expect(parseLinkAndReturn(locationUrl, 'http://server/pre///other/path')).toEqual('http://server/pre///other/path');
```

Comment:
It's not just multiple backslashes. Basically `\s*[\\/]{2,}` would be a problem (but since in parseAppUrl `/` is prepended, the whitespace vector goes away, so I think changing:

```for (var i = 0; url.charAt(i) === '/'; i++)```
to 
```for (var i = 0; url.charAt(i) === '/' || url.charAt(i) === '\\' ; i++)```

would be enough to fix the bug. 

That said, I don't know if the multiple slashes are worth preserving in the beginning of a path for Angular apps. Existing apps served on those would be broken without this patch, so there is no backwards-compatibility problem and it's hard to tell if accepting "//" would not cause further errors when resolving relative URLs for the app later on. Prohibiting that overall seems like a good, conservative choice.